### DISCUSSION TITLE
ux_icon via php

### DISCUSSION THREAD

**Discussion by tacman** [symfony/ux#2421] [Questions & Answers] [answered]
Is there a way to get the icon via a php function, similar to ux_icon(icon_name) in twig?

I'm creating my menus in PHP, e.g.

```php
        $this->add($menu, id: 'gallery', label: 'Slides', icon: 'tabler:images');
        $this->add($menu, id: 'share', label: 'Share', icon: 'tabler:qrcode');
```

In my php code, I'd like to get the svg directly, rather than outputting it in twig as a ux_icon call.  I tried injecting the UXIconRuntime service, but that didn't work.  

Thanks.

**Comment by smnandre** [comment, marked as answer]
Did you try with the IconRendererInterface ?

**Reply by tacman** [reply]
Indeed, that's exactly what I was looking for.  Thanks!
